ElectromagneticModels/coupledElectromagneticModels/externalElectricPotential/externalElectricPotentialFvPatchScalarField.C

```diff
@@ -0,0 +1,355 @@
+    const fvPatch& p,
+    const DimensionedField<scalar, volMesh>& iF,
+    const fvPatchFieldMapper& mapper
+)
+:
+    mixedFvPatchScalarField(ptf, p, iF, mapper),
+    haveI_(ptf.haveI_),
+    I_(ptf.I_),
+    haveJ_(ptf.haveJ_),
+    J_(haveJ_ ? mapper(ptf.J_)() : scalarField()),
+    havePot_(ptf.havePot_),
+    ePotExt_(ptf.ePotExt_, false)
+    //thicknessLayers_(ptf.thicknessLayers_),
+    //kappaLayers_(ptf.kappaLayers_)
+{}
+
+
+Foam::externalElectricPotentialFvPatchScalarField::
+externalElectricPotentialFvPatchScalarField
+(
+    const externalElectricPotentialFvPatchScalarField& tppsf,
+    const DimensionedField<scalar, volMesh>& iF
+)
+:
+    mixedFvPatchScalarField(tppsf, iF),
+    haveI_(tppsf.haveI_),
+    I_(tppsf.I_),
+    haveJ_(tppsf.haveJ_),
+    J_(tppsf.J_),
+    havePot_(tppsf.havePot_),
+    ePotExt_(tppsf.ePotExt_, false)
+    //thicknessLayers_(tppsf.thicknessLayers_),
+    //kappaLayers_(tppsf.kappaLayers_)
+{}
+
+
+// * * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * //
+
+void Foam::externalElectricPotentialFvPatchScalarField::map
+(
+    const fvPatchScalarField& ptf,
+    const fvPatchFieldMapper& mapper
+)
+{
+    mixedFvPatchScalarField::map(ptf, mapper);
+
+    const externalElectricPotentialFvPatchScalarField& tiptf =
+        refCast<const externalElectricPotentialFvPatchScalarField>(ptf);
+
+    if (haveJ_)
+    {
+        mapper(J_, tiptf.J_);
+    }
+/*
+    if (havePot_)
+    {
+        mapper(ePotExt_, tiptf.ePotExt_);
+    }
+*/
+}
+
+
+void Foam::externalElectricPotentialFvPatchScalarField::reset
+(
+    const fvPatchScalarField& ptf
+)
+{
+    mixedFvPatchScalarField::reset(ptf);
+
+    const externalElectricPotentialFvPatchScalarField& tiptf =
+        refCast<const externalElectricPotentialFvPatchScalarField>(ptf);
+
+    if (haveJ_)
+    {
+        J_.reset(tiptf.J_);
+    }
+/*
+    if (havePot_)
+    {
+        ePotExt_.reset(tiptf.ePotExt_);
+    }
+*/
+}
+
+
+void Foam::externalElectricPotentialFvPatchScalarField::updateCoeffs()
+{
+    if (updated())
+    {
+        return;
+    }
+
+    const scalarField& ePotP(*this);
+
+    const electromagneticModel& em =
+        patch().boundaryMesh().mesh()
+       .lookupType<electromagneticModel>();
+
+    const scalarField sigma(em.sigma(patch().index()));
+    const scalar ePotExt = ePotExt_->value(this->db().time().userTimeValue());
+
+    // Compute the total non-convective heat flux
+    scalarField ePotTot(ePotP.size(), 0);
+    if (haveI_)
+    {
+        ePotTot += I_/gSum(patch().magSf())/snGrad()/sigma;
+    }
+    if (haveJ_)
+    {
+        ePotTot += J_/snGrad()/sigma;
+    }
+    if (!havePot_)
+    {
+        ePotTot += ePotExt;
+    }
+
+    // Evaluate
+    //if (!haveh_)
+    //{
+        refValue() = ePotTot;
+        //Switch to zero gradient condition if sigma->0
+        valueFraction() = sigma/(sigma + SMALL);
+        //refGrad() = 0;
+    //}
+/*
```

Comment:
If there is a reason to keep this block of comment, maybe add some notes on why it might be useful